### PR TITLE
chore(v2): use k8s client to get minio credential and record aws region in mlmd …

### PR DESCRIPTION
**Pull request by capri-xiyue**
use k8s client to get minio credential and record aws region in mlmd artifact custom props

**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by capri-xiyue** [comment]
Screen Shot 2021-04-21 at 5 01 08 AM
s3_region custom property gets recorded

File: v2/component/launcher.go [hunk 1]
```diff
@@ -712,8 +716,14 @@ func getExecutorOutput() (*pipeline_spec.ExecutorOutput, error) {
 
 func (l *Launcher) openBucket() (*blob.Bucket, error) {
 	if l.bucketConfig.scheme == "minio://" {
+		secret, err := getMinioCredential()
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get minio credential: %v", err)
+		}
+		accessKey := string(secret.Data["accesskey"])
+		secretKey := string(secret.Data["secretkey"])
```
Comment (Bobgy) [inline v2/component/launcher.go:724]: Report an error when secret doesn't contain these keys?

File: v2/component/launcher.go [hunk 2]
```diff
@@ -712,8 +716,14 @@ func getExecutorOutput() (*pipeline_spec.ExecutorOutput, error) {
 
 func (l *Launcher) openBucket() (*blob.Bucket, error) {
 	if l.bucketConfig.scheme == "minio://" {
+		secret, err := getMinioCredential()
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get minio credential: %v", err)
```
Comment (Bobgy) [inline v2/component/launcher.go:721]:
```suggestion
			return nil, fmt.Errorf("Failed to get minio credential: %w", err)
```

use `%w` to wrap errors?

Similarly for other cases, I think we typically prefer `%w` over '%v' for errors.

File: v2/component/launcher.go [hunk 3]
```diff
@@ -727,3 +737,17 @@ func (l *Launcher) openBucket() (*blob.Bucket, error) {
 	}
 	return blob.OpenBucket(context.Background(), l.bucketConfig.bucketURL())
 }
+
+func getMinioCredential() (*v1.Secret, error) {
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize kubernetes client: %v", err)
```
Comment (Bobgy) [inline v2/component/launcher.go:744]: %w

File: v2/component/launcher.go [hunk 4]
```diff
@@ -727,3 +737,17 @@ func (l *Launcher) openBucket() (*blob.Bucket, error) {
...
+	}
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize kubernetes client set: %v", err)
```
Comment (Bobgy) [inline v2/component/launcher.go:748]: %w

File: v2/component/launcher.go [hunk 5]
```diff
@@ -727,3 +737,17 @@ func (l *Launcher) openBucket() (*blob.Bucket, error) {
...
+	}
+	namespace := os.Getenv("KFP_NAMESPACE")
```
Comment (Bobgy) [inline v2/component/launcher.go:750]: nit return error when env not exist or is empty?

**Review by Bobgy — COMMENTED** [review]
/lgtm
only some nit pickings

**Comment by Bobgy** [comment]
/approve
You can self LGTM after fixing these comments

File: v2/component/launcher.go (hunk 5 above)
Comment (capri-xiyue) [reply v2/component/launcher.go:750]: done

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you Xiyue!

**Pull request merged**